lib/utils.coffee

```diff
@@ -39,9 +40,16 @@ setTabIndex = (elems) ->
 # ==================================================
 # Template
 #
+dirTemplate = (directory, date, editor) ->
+  if editor
+    @frontMatter = new FrontMatter(editor)
+    title = dasherize(@frontMatter.get("title") || getTitleSlug(editor.getPath()))
+    category = @frontMatter.get("category")
+  info =
+    title: title || dasherize("New Post")
+    category: category || "uncategorized"
```

Comment:
This function should process everything automatically rather than the caller doing by itself, doesn't it? If not, it will be just a little extension for `template`. If we add other features like move a post to its proper post dir, we could merely pass the `editor` to it, which i think is easier.